### DISCUSSION TITLE
xattr_compat help

### DISCUSSION THREAD

**Discussion by jr0dd** [openzfs/zfs#12619] [Q&A] [answered]
I was moving a pool from TrueNAS SCALE to Ubuntu 21.04 and can not import the pool. I compiled 2.1.99 from source and not sure if they added some vendor locking junk.

```
This pool uses the following feature(s) not supported by this system:
	com.ixsystems:xattr_compat (Support for cross-platform compatible xattr namespace encoding)
```

**Comment by ghost** [comment, marked as answer]
We have been trying to upstream this feature. It is not intended to lock you in but rather to facilitate migration of pools between operating systems. Unfortunately it did not receive much review until recently. The feature is only present in the nightly/beta TrueNAS SCALE builds and was not activated by default. Unfortunately you will have to destroy the datasets using this feature in order to disable it, either in TrueNAS or by building our [truenas/zfs-2.1-release](https://github.com/truenas/zfs/tree/truenas/zfs-2.1-release) ZFS branch on Ubuntu.

**Reply by jr0dd** [reply]
Thanks. I was chatting with Kris about it in the discord. I'm just gonna build your debs for now and replicate the datasets locally over time unless this gets merged upstream sooner.

**Reply by jr0dd** [reply]
well unfortunately there is no disabling this after pool creation. this is an absolute disaster.
